Assets/MixedRealityToolkit/Utilities/Editor/PackageManifest/PackageManifestUpdater.cs

```diff
@@ -42,10 +46,96 @@ private static string GetPackageManifestFilePath()
             return manifestPath;
         }
 
+        internal static bool TryGetVersionComponents(
+            string packageVersion,
+            out Version version,
+            out float prerelease)
```

Comment:
FWIW, I don't think the suffix is always going to be a float - I think it's an arbitrary alphanumeric thing right?

A pre-release version number MAY be denoted by appending an arbitrary string immediately following the patch version and a dash. The string MUST be comprised of only alphanumerics plus dash [0-9A-Za-z-]. Pre-release versions satisfy but have a lower precedence than the associated normal version. Precedence SHOULD be determined by lexicographic ASCII sort order. For instance: 1.0.0-alpha1 < 1.0.0-beta1 < 1.0.0-beta2 < 1.0.0-rc1 < 1.0.0.

i.e. it would just be alphabetical sorting after that.

I don't think it's necessarily worth making the change right now but going forward we should just reference the spec where they are spec defined.

Reply:
this is based 100% on the format used by MSBuild for Unity. I fully agree that a more robust system should be explored